dbms/src/Storages/Page/V3/CheckpointFile/CPDataFileReader.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/Allocator.h>
+#include <Storages/Page/Page.h>
+#include <Storages/Page/V3/BlobStore.h>
+#include <Storages/Page/V3/PageEntry.h>
+#include <Storages/Page/V3/Universal/UniversalPageId.h>
+#include <aws/s3/model/GetObjectResult.h>
+
+namespace Aws::S3
+{
+class S3Client;
+}
+
+namespace DB::PS::V3
+{
+using UniversalPageMap = std::map<UniversalPageId, Page>;
+using UniversalPageIdAndEntry = std::pair<UniversalPageId, PS::V3::PageEntryV3>;
+using UniversalPageIdAndEntries = std::vector<UniversalPageIdAndEntry>;
+
+class CPDataFileReader : private Allocator<false>
```

Comment:
Please add documents about this class. The usage and the basic idea behind it.

Reply:
Added.

Reply:
I wonder whether `CPDataFileReader ` is a proper name or not. Because it does not contain any logic about "CheckPoint". It is more like a `S3PageReader`.

Reply:
I think this class can be used to read Pages not only from checkpoints.  For example, we will implement evict local Page data later.

Reply:
Fixed.